ec2.template

```diff
@@ -134,6 +143,26 @@
         },
         ""
       ]
+    },
+    "SubnetLocationPublic": {
+      "Fn::Equals": [
+        {
+          "Ref": "SubnetLocation"
+        },
+        "public"
+      ]
+    },
+    "NotSubnetLocationPublic": {
```

Comment:
For the sake of logical clarity, wouldn't naming this SubnetLocationPrivate be more sensical?